jetty-webapp/src/main/java/org/eclipse/jetty/webapp/WebInfConfiguration.java

```diff
@@ -522,21 +526,30 @@ public void configureTempDirectory(File dir, WebAppContext context)
         if (dir == null)
             throw new IllegalArgumentException("Null temp dir");
 
-        //if dir exists and we don't want it persisted, delete it
-        if (dir.exists() && !context.isPersistTempDirectory())
+        // if dir exists and we don't want it persisted, delete it
+        if (!context.isPersistTempDirectory() && dir.exists() && !IO.delete(dir))
```

Comment:
how hard is it to check if it is empty or not.   if it is empty, it would be good to avoid deleting it.

Reply:
Not to terrible to check if empty.
Let me try.

Reply:
This has been implemented and is available in this PR.